### ISSUE TITLE
L(x, use_list=None) doesn't make a list if x is a numpy.float64

### ISSUE THREAD

**Issue by hal-314** [fastai/fastcore#143]
L(x, use_list=None) behaves differently if you pass a float or a numpy.float64/numpy.float32, etc

To reproduce:
```
print(L(np.asarray([7.,5.], dtype=np.float64)[0]))
print(L(np.asarray([7.,5.], dtype=np.float32)[0], use_list=None))
print(L(7., use_list=None))
```
Prints:
```
[7.0]
7.0
[7.0]
```

I found this issue then using Optuna 2.2 to optimize a model (it didn't happen with Optuna 2.1). Optimizer couldn't set the value. Here is the error stack:
```
> /miniconda3/envs/pytorch/lib/python3.7/site-packages/fastcore/foundation.py(271)__len__()
    269     "Base class for composing a list of `items`"
    270     def __init__(self, items): self.items = items
--> 271     def __len__(self): return len(self.items)
    272     def __getitem__(self, k): return self.items[list(k) if isinstance(k,CollBase) else k]
    273     def __setitem__(self, k, v): self.items[list(k) if isinstance(k,CollBase) else k] = v

ipdb>  self.items

0.00038467256590927996

ipdb>  type(self.items)

<class 'numpy.float64'>

ipdb>  u

> /miniconda3/envs/pytorch/lib/python3.7/site-packages/fastai/optimizer.py(43)set_hyper()
     41             else: v = [v.stop/10]*(len(self.param_lists)-1) + [v.stop]
     42         v = L(v, use_list=None) # <-- Here, v isn't a list or something with len() method
---> 43         if len(v)==1: v = v*len(self.param_lists)
     44         assert len(v) == len(self.hypers), f"Trying to set {len(v)} values for {k} but there are {len(self.param_lists)} parameter groups."
     45         self._set_hyper(k, v)


```

**Comment by Salehbigdeli** [comment]
The problem comes from `_is_array` function, this function checks if given argument is `numpy.ndarray` , `pandas.DataFrame` or `pandas.Series`. But unfortunately it simply checks with attributes not exact `shape` of the given parameter.
I'm going to fix it.

The problem:
```
#export
def _is_array(x): return hasattr(x,'__array__') or hasattr(x,'iloc')
```

My solution:
```
#export
def _is_array(x): 
    if hasattr(x,'__array__') or hasattr(x,'iloc'): return bool(x.shape)
```

**Comment by hal-314** [comment]
To give some context, I discovered this bug when using Optuna 2.2 experimental multivariate TPESampler with fastai. In 2.3, it returns a [numpy value instead of float](https://github.com/optuna/optuna/issues/1952). So, I couldn't train the network. Next Optuna version has fixed.

**Comment by Salehbigdeli** [comment]
Glad to see a fix for it. But, what do you think about this situation? Isn't it better to create a list when input is a `numpy.array`?

**Comment by hal-314** [comment]
Yes. I'd expect [basic numpy types](https://numpy.org/doc/stable/user/basics.types.html) as numpy.float64 to behave like basic python types as a int/float when they are passed to `L`. 
```
L(2, use_list=None) # -> (#1) [2]
L(3.4, use_list=None)  # -> (#1) [3.4]
L(np.float32(1.0), use_list=None) # It should be -> (#1) [1.] instead of 1.
```

**Comment by hamelsmu** [comment]
This looks like it was closed via #146
